src/Domain/Creators/SphericalShells.hpp

```diff
@@ -0,0 +1,260 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <string>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+#include "Domain/BoundaryConditions/BoundaryCondition.hpp"
+#include "Domain/BoundaryConditions/GetBoundaryConditionsBase.hpp"
+#include "Domain/CoordinateMaps/Distribution.hpp"
+#include "Domain/Creators/DomainCreator.hpp"
+#include "Domain/Creators/TimeDependence/TimeDependence.hpp"
+#include "Domain/Creators/TimeDependentOptions/Sphere.hpp"
+#include "Options/Auto.hpp"
+#include "Options/Context.hpp"
+#include "Options/String.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <size_t Dim, typename T>
+class DirectionMap;
+template <size_t Dim>
+class Domain;
+namespace domain {
+namespace CoordinateMaps {
+template <size_t Dim>
+class Identity;
+class Interval;
+template <typename Map1, typename Map2>
+class ProductOf2Maps;
+class SphericalToCartesianPfaffian;
+}  // namespace CoordinateMaps
+
+template <typename SourceFrame, typename TargetFrame, typename... Maps>
+class CoordinateMap;
+}  // namespace domain
+/// \endcond
+
+namespace domain::creators {
+/// \brief A set of concentric spherical shells
+
+/// \note This domain will use a spherical harmonic basis in the angular
+/// directions.  It cannot be used with subcell.
+///
+/// \see Sphere for a spherical domain compatible with subcell
+///
+/// This domain creator offers one grid anchor "Center" at the origin.
+///
+/// #### Time dependent maps
+/// There are two ways to add time dependent maps to the SphericalShells domain
+/// creator. In the input file, these are specified under the
+/// `TimeDependentMaps:` block.
+///
+/// ##### TimeDependence
+/// You can use a simple TimeDependence (e.g.
+/// `domain::creators::time_dependence::UniformTranslation` or
+/// `domain::creators::time_dependence::RotationAboutZAxis`) to add time
+/// dependent maps. This method will add the same maps to all blocks in the
+/// domain.
+///
+/// ##### Hard-coded time dependent maps
+/// The SphericalShells domain creator also has the option to use some hard
+/// coded time dependent maps that may be useful in certain scenarios. This
+/// method adds the maps in `domain::creators::sphere::TimeDependentMapOptions`
+/// to the domain. Currently, the first (inner-most) shell has maps between
+/// `Frame::Grid`, `Frame::Distorted`, and `Frame::Inertial` while all
+/// subsequent shells only have maps between `Frame::Grid` and
+/// `Frame::Inertial`.
+///
+/// ##### None
+/// To not have any time dependent maps, pass a `std::nullopt` as the
+/// appropriate argument in the constructor. In the input file, simply have
+/// `TimeDependentMaps: None`.
+class SphericalShells : public DomainCreator<3> {
```

Comment:
Drive by comment. I don't think I'll hold up this PR over this, but I strongly recommend adding spherical shells to the current Sphere domain, rather than creating a new domain creator. This is twofold

1. Maintainability. There are a lot of features in the Sphere domain that we'd have to keep feature parity with this new creator.
2. Interoperability. Long term, I'd like to mix and match spherical shells with cubed shells. Even if right now we only require one or the other for the entire domain. It'll be easier to extend if everything is in one domain creator.

Let me know if there were discussions offline about this design choice, as I could just be out of the loop.

Reply:
I agree with Kyle. What is your reasoning for making it separate? Or maybe it's easier to just discuss after the spectre call on Monday. I've reviewed the rest of the PR, so if you want to split and get that merged first, I'm fine with that.

Reply:
- For just S2 spherical shells, this domain creator is simple and doesn't have to deal with multiple options that make no sense for such a domain.
- I will modify the existing domain when I get non-conforming shells working so that it can have an outer layer of spherical shells.